src/include/platform/PersistedStorage.h

```diff
@@ -87,6 +91,139 @@ CHIP_ERROR Read(Key aKey, uint32_t & aValue);
  */
 CHIP_ERROR Write(Key aKey, uint32_t aValue);
 
+class KeyValueStore
+{
+public:
+    virtual ~KeyValueStore() = default;
+    /**
+     * @brief
+     *   Reads the value of an entry in the KVS. The value is read into the
+     *   provided buffer and the number of bytes read is returned. If desired,
+     *   the read can be started at an offset.
+     *
+     *   If the output buffer is too small for the value, Get returns
+     *   CHIP_ERROR_BUFFER_TOO_SMALL with the number of bytes read. The remainder of the
```

Comment:
If `read_bytes_size` is null and `CHIP_ERROR_BUFFER_TOO_SMALL` is returned, can caller assume that `buffer_size` bytes were read?  Whether the answer is yes or no, worth documenting here.

Reply:
Added a comment